src/rules/guard-super-call.ts

```diff
@@ -21,13 +21,18 @@ const rule: Rule.RuleModule = {
     messages: {
       guardSuperCall:
         'Super calls to lifecycle callbacks should be guarded in case the base class does not implement them'
-    }
+    },
+    schema: [{
+      requireTypeCheck: {type: 'boolean'}
+    }]
   },
 
   create(context): Rule.RuleListener {
     let insideNonNativeElement = false;
     let errNode = null;
     const source = context.getSourceCode();
+    const options = context.options?.[0] ?? {};
+    const requireTypeCheck = options.requireTypeCheck ?? false;
```

Comment:
its probably better to just compare here, like `options.requireTypeCheck === true`